_posts/2018-08-26-eager-dcgan/eager-dcgan.Rmd

```diff
@@ -292,30 +294,27 @@ At the start of each epoch, we create a fresh iterator over the dataset:
 
 
 ```{r}
- for (epoch in seq_len(num_epochs)) {
-    start <- Sys.time()
-    total_loss_gen <- 0
-    total_loss_disc <- 0
-    iter <- make_iterator_one_shot(train_dataset)
+for (epoch in seq_len(num_epochs)) {
+  start <- Sys.time()
+  total_loss_gen <- 0
+  total_loss_disc <- 0
+  iter <- make_iterator_one_shot(train_dataset)
 ```
 
 Now for every batch we obtain from the iterator, we are calling the generator and having it generate images from random noise. Then, we're calling the dicriminator on real images as well as the fake images just generated. For the discriminator, its relative outputs are directly fed into the loss function. For the generator, its loss will depend on how the discriminator judged its creations:
 
 ```{r}
- until_out_of_range({
-      batch <- iterator_get_next(iter)
-      noise <- k_random_normal(c(batch_size, noise_dim))
-      with(tf$GradientTape() %as% gen_tape, {
-      with(tf$GradientTape() %as% disc_tape, {
-        generated_images <- generator(noise)
-        disc_real_output <- discriminator(batch, training = TRUE)
-        disc_generated_output <-
-          discriminator(generated_images, training = TRUE)
-        gen_loss <- generator_loss(disc_generated_output)
-        disc_loss <-
-          discriminator_loss(disc_real_output, disc_generated_output)
-      })
-      })
+until_out_of_range({
+  batch <- iterator_get_next(iter)
+  noise <- k_random_normal(c(batch_size, noise_dim))
+  with(tf$GradientTape() %as% gen_tape, { with(tf$GradientTape() %as% disc_tape, {
+    generated_images <- generator(noise)
+    disc_real_output <- discriminator(batch, training = TRUE)
+    disc_generated_output <-
+       discriminator(generated_images, training = TRUE)
+    gen_loss <- generator_loss(disc_generated_output)
+    disc_loss <- discriminator_loss(disc_real_output, disc_generated_output)
+    }) })
```

Comment:
I still think these needs to align w/ the outer `with()`, so e.g.

```r
with(tf$GradientTape() %as% gen_tape, { with(tf$GradientTape() %as% disc_tape, {
    generated_images <- generator(noise)
    disc_real_output <- discriminator(batch, training = TRUE)
    disc_generated_output <-
       discriminator(generated_images, training = TRUE)
    gen_loss <- generator_loss(disc_generated_output)
    disc_loss <- discriminator_loss(disc_real_output, disc_generated_output)
})})
```

Reply:
But with less indentation for the code inside the blocks (I think GH did that somehow on copy/paste)

Reply:
fixing